src/main/java/io/stargate/sgv2/jsonapi/service/schema/collections/DocumentPath.java

```diff
@@ -77,6 +77,42 @@ public String getSegment(int index) {
     return segments.get(index);
   }
 
+  /**
+   * Verifies that the encoded path does not contain any lone ampersands. A lone ampersand is one
+   * that is not followed by another ampersand or a dot, which are the only valid escape sequences.
+   *
+   * @param path the encoded path string to verify
+   * @return the original path if it passes verification
+   * @throws IllegalArgumentException if the path contains a lone ampersand
+   */
+  public static String verifyEncodedPath(String path) {
+    Objects.requireNonNull(path, "path cannot be null");
+
+    for (int i = 0; i < path.length(); i++) {
+      char ch = path.charAt(i);
```

Comment:
Minor tweaks: change `i++` to `i` in loop, use

    char ch = path.charAt(i++);

and later on

    if (i >= path.length()) { ... |

and

     char nextChar = path.charAt(i++);